File: PHP/7php 3/sumCubes.php
<?php
/*
|--------------------------------------------------------------------------
| Sum of Cubes
|--------------------------------------------------------------------------
|
*/

function sumCubes($n) {
   $res = 0;
   for ($i = 1; $i <= $n; $i++) {
      $res += $i ** 3;
   }
   return $res;
}


function sumCubes($n) {
   return $n * $n * ($n + 1) * ($n + 1) / 4;
}


function sumCubes($n) {
   if ($n == 1) return 1;
   return $n*$n*$n + sumCubes($n-1);
}


function sumCubes($n) {
   return array_reduce(range(1, $n), function ($c, $i) {
      return $c + $i**3;
   });
}


print_r(sumCubes(1) . ' ' . 1 . '
');
print_r(sumCubes(2) . ' ' . 9 . '
');
print_r(sumCubes(3) . ' ' . 36 . '
');
print_r(sumCubes(4) . ' ' . 100 . '
');
print_r(sumCubes(10) . ' ' . 3025 . '
');
print_r(sumCubes(123) . ' ' . 58155876 . '
');

File: Python/7py 3/sum_cubes.py
"""
|--------------------------------------------------------------------------
| Sum of Cubes
|--------------------------------------------------------------------------
|
"""

def sum_cubes(n):
   res = 0
   for i in range(1, n+1):
      res += i**3
   return res


def sum_cubes(n):
   return sum(i**3 for i in range(0,n+1))


def sum_cubes(n):
   return (n*(n+1)//2)**2


def sum_cubes(n: int) -> int:
   return sum(list(map(lambda _it: pow(_it, 3), range(n + 1))))


print(sum_cubes(1), 1)
print(sum_cubes(2), 9)
print(sum_cubes(3), 36)
print(sum_cubes(4), 100)
print(sum_cubes(10), 3025)
print(sum_cubes(123), 58155876)